### PR TITLE
Move keytar to an optional dependency

### PR DESCRIPTION
**Pull request by childish-sambino**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -3562,8 +3552,7 @@
     "strip-json-comments": {
       "version": "2.0.1",
       "resolved": "https://registry.npmjs.org/strip-json-comments/-/strip-json-comments-2.0.1.tgz",
-      "integrity": "sha1-PFMZQukIwml8DsNEhYwobHygpgo=",
-      "dev": true
```
Comment (krantitalluri) [inline package-lock.json]: some of these are missing optional. is this intentional?

File: package-lock.json [hunk 2]
```diff
@@ -3833,8 +3822,7 @@
     "util-deprecate": {
       "version": "1.0.2",
       "resolved": "https://registry.npmjs.org/util-deprecate/-/util-deprecate-1.0.2.tgz",
-      "integrity": "sha1-RQ1Nyfpw3nMnYvvS1KKJgUGaDM8=",
-      "dev": true
+      "integrity": "sha1-RQ1Nyfpw3nMnYvvS1KKJgUGaDM8="
```
Comment (krantitalluri) [inline package-lock.json]: missing optional or removed?

Comment (childish-sambino) [reply package-lock.json]: The change for these is just that the `"dev": true` line was dropped. This was a result of dropping `keytar` as a `devDependency`. So the ones that changed are no longer part of a dev dependency tree.

File: package-lock.json (hunk 1 above)
Comment (childish-sambino) [reply package-lock.json]: Just means that they are used by some other dependency which is not optional.

**Pull request merged**